test/parallel/test-http-client-check-http-token.js

```diff
@@ -2,39 +2,28 @@
 const common = require('../common');
 const assert = require('assert');
 const http = require('http');
+const Countdown = require('../common/countdown');
 
 const expectedSuccesses = [undefined, null, 'GET', 'post'];
-let requestCount = 0;
+const expectedFails = [-1, 1, 0, {}, true, false, []];
 
-const server = http.createServer((req, res) => {
-  requestCount++;
-  res.end();
+const countdown =
+  new Countdown(expectedSuccesses.length,
+                common.mustCall(() => server.close()));
 
-  if (expectedSuccesses.length === requestCount) {
-    server.close();
-  }
-}).listen(0, test);
+const server = http.createServer(common.mustCall((req, res) => {
+  res.end();
+  countdown.dec();
+}, expectedSuccesses.length));
 
-function test() {
-  function fail(input) {
+server.listen(0, common.mustCall(() => {
+  expectedFails.forEach((method) => {
     assert.throws(() => {
-      http.request({ method: input, path: '/' }, common.mustNotCall());
+      http.request({ method: method, path: '/' }, common.mustNotCall());
```

Comment:
this can be a shorthand. `{ method, path: '/'}`. Also this test structure is a bit strange - and I'm not sure that in this case the code is clearer after the refactoring. Personally I would prefer a `for... of` but it's not a big deal.